File: lib/sphere.py
from vec3 import vec3


class Sphere:
    def __init__(self, pos=vec3(0, 0, 0), r=1):
        self.pos = pos
        self.r = r

    def log(self):
        return "<Sphere r=[%s] pos=[%s, %s, %s]>".format(
                    self.r,
                    self.pos.x,
                    self.pos.y,
                    self.pos.z
                )

File: src/sphere.rs
use cgmath::Vector3;

pub struct Sphere {
    pub rad: f32,
    pub pos: Vector3<f32>
}

impl Sphere {
    pub fn new (rad: f32, (x, y, z): (f32, f32, f32)) -> Sphere {
        Sphere {
            rad: rad,
            pos: Vector3 { x: x, y: y, z: z }
        }
    }
}
